utils/linux-install-luarocks.sh

```diff
@@ -42,7 +42,7 @@ fi
 make build > build.log 2>&1 || (cat build.log && exit 1)
 sudo make install > build.log 2>&1 || (cat build.log && exit 1)
 cd .. || exit
-rm -rf luarocks-"$LUAROCKS_VER"
+rm -rf luarocks-"$LUAROCKS_VER" luarocks-"$LUAROCKS_VER".tar.gz
```

Comment:
Could we remove it just after untar?

Reply:
> Could we remove it just after untar?

on this way, if `configure` failed, it need to download archive file `v3.8.0.tar.gz` repeatedly.

Reply:
OK. So that we can also handle https://github.com/apache/apisix/issues/6307 in this PR.

Reply:
no, #6307 occurs at `configure` failure, so download twice.

Reply:
Yes. But they change the same thing twice. So why not put together?

Reply:
ok, remove it behind untar was better, I'll update this logic.